### PR TITLE
Add combine mul/div by left shifted 1

### PR DESCRIPTION
**Pull request by leonardoalt**
Proves https://github.com/ethereum/solidity/pull/6756

### CODE REVIEW

File: rules/combine_div_shl_one_32.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# Constants
+bv_one = BitVecVal(1, n_bits)
+bv_zero = BitVecVal(0, n_bits)
+
+# Requirements
+
+# Non optimized result
+# Division by 0 is undefined for BitVectors, so we define it.
+nonopt = If(
```
Comment (ekpyron) [inline rules/combine_div_shl_one_32.py]: Actually, we might want to define all opcodes (including stuff like ``SHL`` which matches ``<<`` but one always needs to take care of argument order) as python functions with exactly the name of the opcode in a separate file and import that into the proofs... it'd be nice to have syntactically identical expression in the proofs and I think that should be possible!

Comment (leonardoalt) [reply rules/combine_div_shl_one_32.py]: Yeah, that'd be better

**Comment by leonardoalt** [comment]
Mul didn't even work with 128 =/

File: rules/combine_div_shl_one_32.py (hunk 1 above)
Comment (leonardoalt) [reply rules/combine_div_shl_one_32.py]: I'll merge this as is and then create another PR with the opcodes

**Pull request merged**